src/Components/Components/src/Forms/EditContextExpressionExtensions.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+
+namespace Microsoft.AspNetCore.Components.Forms
+{
+    /// <summary>
+    /// Provides extension methods to simplify using <see cref="EditContext"/> with expressions.
+    /// </summary>
+    public static class EditContextExpressionExtensions
```

Comment:
Is there a really good reason to make these extension methods? Would anything go wrong if they were just defined on the `EditContext`?

Reply:
It seems like the idea to is just to keep these separated based on how critical they are to the core system's layering.

Reply:
It's a combination of:

 * Making it easier to reason about the correctness of the core system (specifically, its inability to be put into an invalid state by any given API call). Anything expressed as extension methods is safe because it doesn't change the true API surface.
 * Proving that certain types of extensibility are possible by external code. For example, the DataAnnotations stuff being extension methods is to ensure we're not relying on any private APIs. (I know we could still make the mistake of relying on an `internal`, but that one thing is easier to spot than if everything was in one file).

TBH with the `EditContextExpressionExtensions` cases it doesn't really matter very much, but I'm inclined to leave it as-is unless there are specific objections.

Reply:
No, I think these are all fine reasons.